sw/host/opentitanlib/src/console/spi.rs

```diff
@@ -46,8 +78,33 @@ impl<'a> SpiConsoleDevice<'a> {
         // Read the SPI console frame data.
         let data_len_bytes_w_pad = (data_len_bytes + 3) & !3;
         let mut data = vec![0u8; data_len_bytes_w_pad];
-        self.spi
-            .run_transaction(&mut [Transfer::Write(&[0xff; 4]), Transfer::Read(&mut data)])?;
+        let data_address: u32 = (read_address
+            + u32::try_from(SpiConsoleDevice::SPI_FRAME_HEADER_SIZE).unwrap())
+            % SpiConsoleDevice::SPI_FLASH_READ_BUFFER_SIZE;
+
+        // Handle wrap-around
+        if (SpiConsoleDevice::SPI_FLASH_READ_BUFFER_SIZE - data_address)
+            < u32::try_from(data_len_bytes_w_pad).unwrap()
+        {
+            let first_part_size: usize =
+                usize::try_from(SpiConsoleDevice::SPI_FLASH_READ_BUFFER_SIZE - data_address)
+                    .unwrap();
+            self.flash
+                .read(self.spi, data_address, &mut data[0..first_part_size])?;
+            self.flash.read(
+                self.spi,
+                0,
+                &mut data[first_part_size..data_len_bytes_w_pad],
+            )?;
+        } else {
+            self.flash.read(self.spi, data_address, &mut data)?;
+        }
```

Comment:
consider wrapping in a function since this code block is similar to the above?

Reply:
Good idea! Encapsulated the code snippets into `fn read_data` functions in the latest commit.